jupyterlab/extensions/pypi.py

```diff
@@ -0,0 +1,405 @@
+            if install_metadata["packageManager"] == "python":
+                return self._normalize_name(install_metadata["packageName"])
+        return self._normalize_name(extension.name)
+
+    async def __throttleRequest(self, recursive: bool, fn: Callable, *args) -> Any:
+        """Throttle XMLRPC API request
+
+        Args:
+            recursive: Whether to call the throttling recursively once or not.
+            fn: API method to call
+            *args: API method arguments
+        Returns:
+            Result of the method
+        Raises:
+            xmlrpc.client.Fault
+        """
+        current_loop = tornado.ioloop.IOLoop.current()
+        try:
+            data = await current_loop.run_in_executor(None, fn, *args)
+        except xmlrpc.client.Fault as err:
+            if err.faultCode == -32500 and err.faultString.startswith("HTTPTooManyRequests:"):
+                delay = 1.01
+                match = re.search(r"Limit may reset in (\d+) seconds.", err.faultString)
+                if match is not None:
+                    delay = int(match.group(1) or "1")
+                self.log.info(
+                    f"HTTPTooManyRequests - Perform next call to PyPI XMLRPC API in {delay}s."
+                )
+                await asyncio.sleep(delay * self.rpc_request_throttling + 0.01)
+                if recursive:
+                    data = await self.__throttleRequest(False, fn, *args)
+                else:
+                    data = await current_loop.run_in_executor(None, fn, *args)
+
+        return data
+
+    async def list_packages(
+        self, query: str, page: int, per_page: int
+    ) -> Tuple[Dict[str, ExtensionPackage], Optional[int]]:
+        """List the available extensions.
+
+        Note:
+            This will list the packages based on the classifier
+                Framework :: Jupyter :: JupyterLab :: Extensions :: Prebuilt
+
+            Then it filters it with the query
+
+            We do not try to check if they are compatible (version wise)
+
+        Args:
+            query: The search extension query
+            page: The result page
+            per_page: The number of results per page
+        Returns:
+            The available extensions in a mapping {name: metadata}
+            The results last page; None if the manager does not support pagination
+        """
+        matches = await self.__get_all_extensions()
+
+        extensions = {}
+
+        for name, group in groupby(filter(lambda m: query in m[0], matches), lambda e: e[0]):
+            _, latest_version = list(group)[-1]
+            data = await _fetch_package_metadata(name, latest_version, self.base_url)
+
+            normalized_name = self._normalize_name(name)
+            extensions[normalized_name] = ExtensionPackage(
+                name=normalized_name,
+                description=data.get("summary"),
+                url=data.get("home_page", data.get("package_url")),
```

Comment:
I will display only one URL - targeting the homepage. It may be confusing for the user to get two links. And the homepage should be the better one to accompany a new user.